cpp/dolfinx/fem/FiniteElement.cpp

```diff
@@ -292,3 +296,39 @@ bool FiniteElement::needs_permutation_data() const noexcept
   return _needs_permutation_data;
 }
 //-----------------------------------------------------------------------------
+void FiniteElement::map_push_forward(
```

Comment:
I's suggest templating these functions on the C++ side, see https://github.com/FEniCS/dolfinx/blob/master/cpp/dolfinx/fem/FiniteElement.h#L187.